tests/python/unittest/test_pass_simplify.py

```diff
@@ -52,8 +54,69 @@ def test_canonical():
     ret2 = tvm.ir_pass.CanonicalSimplify(x % 3 + x % 4)
     assert (tvm.ir_pass.Equal(ret1, ret2))
 
+
+def test_simplify_combiner():
+    dummy = tvm.var('dummy')
+
+    prod = comm_reducer(lambda x, y: x*y, lambda t0: tvm.const(1, t0))
+
+    sum_and_prod = comm_reducer(lambda x, y: (x[0] + y[0],
+                                              x[1]*y[1]),
+                                lambda t0, t1: (tvm.const(0, t0),
+                                                tvm.const(5, t0) - tvm.const(4, t0)))
+
+    sum_or_prod = comm_reducer(lambda x, y: tvm.select(dummy < 0, x + y, x*y),
+                               lambda t0: tvm.select(dummy < 0, tvm.const(0, t0), tvm.const(1, t0)))
+
+    sum_and_prod2 = comm_reducer(lambda x, y: (x[0] + y[0],
+                                               x[1]*y[1] + 0*x[0] + y[0] - y[0]),
+                                 lambda t0, t1: (tvm.const(5, t0) - tvm.const(5, t0),
+                                                 tvm.const(1, t1)))
+
+    some_reducer1 = comm_reducer(lambda x, y: (x[0] + y[0],
+                                               x[0] + y[0] + x[1] + y[1],
+                                               x[0]*y[2] + y[0]*x[2],
+                                               x[1] + y[2],
+                                               4.0),
+                                 lambda t0, t1, t2, t3, t4: (tvm.const(0, t0),
+                                                             tvm.const(1, t1),
+                                                             tvm.const(2, t2),
+                                                             tvm.const(3, t3),
+                                                             tvm.const(4, t4)))
+
+    k = tvm.reduce_axis((0, 10), name="k")
+    A = tvm.placeholder((10,), name='A')
+
+    assert Equal(Simplify(sum_and_prod((A[k], A[10-k]), k)[0]), tvm.sum(A[k], k))
+    assert Equal(Simplify(sum_and_prod((A[k], A[10-k]), k)[1]), prod(A[10-k], k))
+
+    vrange = {dummy: tvm.Range(-10, -5)}
+    assert Equal(Simplify(sum_or_prod(A[k], k), vrange), tvm.sum(A[k], k))
+    vrange = {dummy: tvm.Range(5, 10)}
+    assert Equal(Simplify(sum_or_prod(A[k], k), vrange), prod(A[k], k))
+
+    assert Equal(Simplify(sum_and_prod2((A[k], A[10-k]), k)[0]), tvm.sum(A[k], k))
+    assert Equal(Simplify(sum_and_prod2((A[k], A[10-k]), k)[1]), prod(A[10-k], k))
+
+    assert [len(Simplify(some_reducer1((A[k], A[10-k], A[0], A[k+1], A[k]), k)[j]).source)
+            for j in range(5)] == [1, 2, 2, 4, 1]
```

Comment:
It's hard to understand for me what this testcase is testing. Can you please add some comments here?

Reply:
+1 Trying to understand the test case, is this testcase also order dependent? 

What's the out of this sequence?
````
(A[k], A[k], A[10-k], A[0], A[k+1])
````

Reply:
No, the sources are completely dummy, they may all be `A[k]` or some constant. The purpose of the test case is to check that only the requested component of the tuple and the component's dependencies are left, given a multi-output reducer with complex dependencies.

Reply:
Thanks.

Instead of checking `source` length, I would suggest check `source` directly, I think that helps other people understand the intention of the testcase, like:
````
# this code won't work, Use ir_pass.Equal to compare expected and actual result
assert [Simplify(some_reducer1((A[k], A[k], A[k], A[k], A[k]), k)[j]).source
        for j in range(5)] == [[A(k)],
                               [A(k), A((10 - k))],
                               [A(k), A(0)],
                               [A(k), A((10 - k)), A(0), A((k + 1))],
                               [A(k)]]
````

Reply:
fixed